tests/integration_tests/functional/test_pause_resume.py

```diff
@@ -1,65 +1,111 @@
 # Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
-"""Tests scenarios for pausing/resuming a microVM."""
-
-import host_tools.network as net_tools
-
-
-def test_pause_resume(test_microvm_with_ssh, network_config):
-    """Test pausing and resuming the vCPUs."""
-    test_microvm = test_microvm_with_ssh
-    test_microvm.spawn()
-
-    # Set up the microVM with 2 vCPUs, 256 MiB of RAM and a root file
-    # system with the rw permission.
-    test_microvm.basic_config()
-
-    # Create tap before configuring interface.
-    _tap1, _, _ = test_microvm.ssh_network_config(network_config, '1')
+"""Basic tests scenarios for snapshot save/restore."""
+import logging
+import platform
+import pytest
+from conftest import test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.matrix import TestMatrix, TestContext
+from framework.builder import MicrovmBuilder
+import host_tools.network as net_tools  # pylint: disable=import-error
+
+
+def _test_pause_resume(context):
+    logger = context.custom['logger']
+    vm_builder = context.custom['builder']
+
+    logger.info("Testing microvm: \"{}\" with kernel {} and disk {} "
+                .format(context.microvm.name(),
+                        context.kernel.name(),
+                        context.disk.name()))
+
+    microvm = vm_builder.build(context.kernel,
+                               [context.disk],
+                               context.microvm)
+    tap = microvm.ssh_network_config(context.custom['network_config'], '1')
```

Comment:
It is used, please see below del tap. If I do not explicitly delete it I get these:
```
Cannot open network namespace "82b35ec3-1ffb-4a45-b214-b8aafa52af45": No such file or directory

Returned error code: 1
stderr:
Cannot open network namespace "82b35ec3-1ffb-4a45-b214-b8aafa52af45": No such file or directory
```

Reply:
Yep, `del _tap` is needed indeed, but it looks like if we only use that variable for deleting it later, python considers `tap` as unused and marks it with that gray annoying highlight, so that's why I've suggested to rename it to `_tap`.